frontend/packages/topology/src/components/quick-search/QuickSearchDetails.tsx

```diff
@@ -25,7 +26,13 @@ const QuickSearchDetails: React.FC<QuickSearchDetailsProps> = ({ selectedItem })
         className="odc-quick-search-details__form-button"
         onClick={() => {
           const { href, callback } = selectedItem.cta;
-          callback ? callback() : history.push(href);
+          if (callback) {
+            callback();
+            removeQueryArgument('catalogSearch');
```

Comment:
You should not remove query arguments directly like this because it breaks the use case where if user clicks on back button on browser the modal will not open with previous search result.

Reply:
> Good catch. That looks like there's also an existing bug.
> How to reproduce:
> 
> * open quick search
> * enter a term
> * click away from the modal to dismiss it
> * Observe: URL still contains query param: `catalogSearch=`
> * refresh the page
> * Observe: topology loads with quick search open
> 
> That being said. I think clicking the `start` button of a quick start should clear the search string and close the modal. This should be fixed as part of this issue.

@rohitkrai03 The idea is to remove the QueryArg in case of a click/callback of QuickStart and not for the case of going for an installation / catalog operation. It will still contain the queryparam

Reply:
@abhinandan13jan The UX designs had this use case that if user goes away to some other page clicking on some action on the quick search modal and clicks back button on the browser, he should be brought back to topology with the modal open with the same search term as before. This use case will be broken by your changes.

Reply:
@rohitkrai03 in this case it makes sense since the query argument will only be removed when there is a callback.